### PR TITLE
Make URL events asynchronous

### LINKED ISSUE #286: Make title module asynchronous

### PR DESCRIPTION
**Pull request by georgewatson**
Resolves #286

### CODE REVIEW

File: modules/url-events.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  //
+}
+
+var getTitle = function (url) {
+  return new Promise(
+    (resolve, reject) => {
+      const maxSize = 1024
+      var size = 0
+      var response = ''
+      var req = request(url)
```
Comment (zuzak) [inline modules/url-events.js]: Can this use `request-promise`?

**Comment by zuzak** [comment]
might be worth rebasing #285 into this

File: modules/url-events.js (hunk 1 above)
Comment (georgewatson) [reply modules/url-events.js]: It quite possibly can, but I tried and never found a way to monitor the download size.

**Comment by georgewatson** [comment]
Resolves #285

**Pull request merged**